app/services/ncr/approval_manager.rb

```diff
@@ -50,7 +50,14 @@ def force_approvers(users)
       individuals = users.map do |user|
         proposal.existing_step_for(user) || Steps::Approval.new(user: user)
       end
-      proposal.root_step = Steps::Serial.new(child_steps: individuals)
+      unless proposal.root_step && steps_share_parent?(proposal.root_step, individuals)
```

Comment:
good point. I have added test and renamed method/vars for clarity in 4cbb1fc

The issue is that, previously, a new root step (`Steps::Serial`) was created every time the WorkOrder was modified, regardless of whether the child steps changed at all. This created a lot of noise in the proposal history, with `parent_id changed` type messages.